GitHub WebHook: include full repository name in repository event notifications

tschneidereit opened the pull request:
This changes the notification messages for events that currently just include the string `"the repository"` to also include the full (`org/repo`) name of the affected repository. Messages for the following events are changed:
- `public`
- `star`
- `watch`
- `repository`
- `team_add`

Background: we're using the GitHub integration for org-wide notifications for the [Bytecode Alliance Zulip](bytecodealliance.zulipchat.com/), and having all messages just say "the repository" isn't ideal. Even now one can hover over the link to see the repo's url, but it'd be much nicer if the message just contained the full name.

I also changed the message for `star` to include a link to the repository, same as the `watch` notification. It might make sense to just use the same notification for both — let me know if you agree and I'm happy to add a patch making that change, too.

<!-- What's this PR for?  (Just a link to an issue is fine.) -->


**Testing Plan:** <!-- How have you tested? -->
I tried setting up a local dev environment using Vagrant, but failed to do so, so I haven't done more than manual verification. I'd be grateful if someone with an existing dev environment could run the tests!

**GIFs or Screenshots:** <!-- If a UI change.  See:
  https://zulip.readthedocs.io/en/latest/tutorials/screenshot-and-gif-software.html
  -->
See above: I wasn't able to set up a dev environment, so can't provide screenshots.

<!-- Also be sure to make clear, coherent commits:
  https://zulip.readthedocs.io/en/latest/contributing/version-control.html
  -->

tschneidereit commented:
Looks like this is now passing all tests — apologies for triggering altogether too many CI runs.

timabbott commented:
@Hypro999 can you review this?  I'm supportive of the change (I agree there's no reason to say "the repository" when we can name it specifically!), though I'm not sure if we want to be putting the repository names in code blocks or just stating them, which might look cleaner.

tschneidereit commented:
> I'm not sure if we want to be putting the repository names in code blocks or just stating them, which might look cleaner.

I'm happy to change that, if you prefer. I went by some other examples, like [the one for the `team` event](https://github.com/zulip/zulip/blob/31d50ef359b40d7ddb9af1d49adbd2330f5613b2/zerver/webhooks/github/view.py#L258), but there are counterexamples, too.

Hypro999 commented on zerver/webhooks/github/view.py:
```diff
@@ -207,8 +207,9 @@ def get_push_commits_body(payload: Dict[str, Any]) -> str:
 
 def get_public_body(payload: Dict[str, Any]) -> str:
-    return "{} made [the repository]({}) public.".format(
+    return "{} made [the repository `{}`]({}) public.".format(
```
We're hyperlinking too much text here. Instead of
```
[the repository `{}`]({})
```
Let's go with:
```
the repository [{}]({})
```

For example, "the repository [bytecodealliance/wasmtime](https://github.com/bytecodealliance/wasmtime)" looks better than "[the repository bytecodealliance/wasmtime](https://github.com/bytecodealliance/wasmtime)".

tschneidereit replied:
Makes sense, and I changed things accordingly.

tschneidereit commented:
@Hypro999 thank you for the prompt review and thoughtful feedback, much appreciated!

> The first three commits don't pass the CI tests but the last one does. Each commit should pass all of the CI tests independently.

My bad, I'm so used to working in repos that squash all PRs that I didn't think of squashing on my end. Fixed now.

I also fixed the linter errors, so things should hopefully be good now.

timabbott commented:
Looks great, merged, thanks @tschneidereit!

tschneidereit commented:
@Hypro999 & @timabbott, thank you both for making this a great experience!

timabbott commented:
@tschneidereit thanks for the kind words and the contribution!  We put a lot of effort into making the contributing experience nice :)
